### DISCUSSION TITLE
3.08c - issue with URL channel change

### DISCUSSION THREAD

**Discussion by rblancarte** [GerryDazoo/Slinger#161] [General]
I downloaded and setup v3.08c last night. Works great, except:

when using the URL channel change, the button presses don't get sent quickly enough and cause my Tivo to not get set to the right channel. An example of this video is up on Youtube here:

https://www.youtube.com/watch?v=Ps7dq70e8h4

I try to tune channels 510 and 574, you can see how it misses the channel. One time (I didn't catch it on video) I tried to tune 510 and I got channel 5 then channel 10.

Anyway thoughts?

My setup is a Raspberry Pi 1, the only thing running is Slinger.

**Comment by rblancarte** [comment]
BTW - one thing I thought about after uploading and posting this - this is only a quick test I ran with my phone using VLC. But this happens with any video player I have tried - MPV, VLC, Plex (via the xteve setup seen on reddit).

**Reply by GerryDazoo** [reply]
please run your test without the xteve setup. And post all of the server output so I can see what's going on.

**Reply by rblancarte** [reply]
So all of the tests this morning (in the video) were without xteve running on top of this. Find attached a log, it's kind of big.

[sling.log](https://github.com/GerryDazoo/Slinger/files/9842165/sling.log)

**Reply by rblancarte** [reply]
Taking this a step further - the connections that I ran this morning were from 10.0.1.246 - that's my iPhone's IP address.

10/21/2022, 09:49:23   Streaming connection from ('10.0.1.246', 64017) -  that's the first one

**Reply by rblancarte** [reply]
and my config.ini

```
[SLINGBOX]
sbtype="350/500/M1/M2"
password=xxxxxxxxxxxxxxx
;ipaddress=192.168.117.110
;port=5201
;finderid=BAD95DAED609364C8204F2BC5FBDC363
;name=something
;Resolution=12
Resolution=16
;FrameRate=30
;VideoBandwidth=2000
;VideoSmoothness=63
;IframeRate=5
;AudioBitRate=32
VideoSource = 1

[SERVER]
port=8080
;maxstreams=10
enableremote=yes

[REMOTE]
code=1
include=./Slinger/CustomRemotes/Tivo.Mini.Remote.txt 

```

**Comment by rblancarte** [comment]
Hmmm. I'm wondering if this issue isn't with my Slingbox and has nothing to do with this script.

I just installed Slinger on my FreeBSD server (in a jail) and it still does this slow remote thing when using the `?channel=123` string.

**Comment by GerryDazoo** [comment]
Can you be a little more specific about what you mean by "slow"? 
Does the channel change correctly?
Don't forget your media player depending on how it's configured can add multiple seconds of delay because of internal buffering before you see the results of the channel change on the video stream.
If you can, watch the video directly from your cable box you should see it change quickly.

**Reply by rblancarte** [reply]
The channels are not changing correctly. 
The YouTube video shows this. 

A description of What’s happening is that I’ll enter a channel in the URL and the slingbox sends these digits one by one with a long delay in them. But the TIVo will many times just flip to the singleton channels because it doesn’t get a second input fast enough.

so an example, the url says channel=123
1 is sent to the tivo 
Long delay
Tivo flips to channel 1
2 is sent to the tivo
Long delay
Tivo changes to channel 2. 
Etc 

I can use the remote control page and it sends the codes fine if I enter them. But the URL method does not.

**Reply by GerryDazoo** [reply]
Please post the server output. The code inserts a 500 mSec delay between key presses. I'm assuming when you say "long" you mean more than 1/2 second. 
You can patch this interval in the python code/ Line 834
                if curtime - last_remote_command_time > 0.5 :
change the 0.5 to a smaller value and see if the helps.
I'm not sure why no one else is having this issue. 
Also, to help see what's really going on can you also change line 882 to
        print(ts(), name,'Sending IR keycode', int(data[0]), rccode, 'for', sender_ip)

**Reply by almoney** [reply]
I am seeing the same issue occasionally.  Some of the slingplayer clients allow you to send a complete channel change as a single request.  I think this would be appropriate for the startup channel.  I also am looking at how to start the "initial" channel with a "Live TV" command first.

**Reply by GerryDazoo** [reply]
Your going to need to wait for the next release, that will allow you to send the "Live TV" keycode before sending the channel change.

**Reply by rblancarte** [reply]
> Please post the server output. 

I will produce some more output on this and post it here. I will run this on the FreeBSD system since that is a little more fresh and the ideal target for what I want to make my overall system.

> The code inserts a 500 mSec delay between key presses. I'm assuming when you say "long" you mean more than 1/2 second.

Yes. It is long enough that the singleton press "times out" and the Tivo changes channels. Running a test on my Tivo Bolt (on Experience 4), this timeout is about 1.5-1.75 seconds long. I will need to run this same test with the Mini soon

> You can patch this interval in the python code/ Line 834 if curtime - last_remote_command_time > 0.5 : change the 0.5 to a smaller value and see if the helps. I'm not sure why no one else is having this issue. Also, to help see what's really going on can you also change line 882 to print(ts(), name,'Sending IR keycode', int(data[0]), rccode, 'for', sender_ip)

I am still wondering if this isn't an issue with the Slingbox itself. I also posted on r/slingbox the audio/video synchronization issue, which I am fairly certain is an issue with my SB hardware (this happens in Slinger and the native SB app). This is why I am contemplating a hard reset of my slingbox (within the next day or 2) and re-setting it up before trying to do more SW debugging on your code. I mean the issue of:
- Being one of the few or only people to see this
- Seeing this happen with different Slinger systems
- Having issues beyond this channel change issue
This all leads me to think it's an issue with the SB and not external SW.

**Reply by rblancarte** [reply]
Yeah, so turning this down from 0.5 to 0.001 and this works. I will keep playing with it. And let you know.

**Comment by rblancarte** [comment]
@GerryDazoo So I just ran a few tests on a FreeBSD server here is the server output. But I also recorded the screen, because it (to me, you might correct me if I am wrong) appears that the button output presses take longer than expected. I'm assuming when I see output on the screen a button press is sent. This appears to be around 2 seconds per press. Thoughts?

Server output: [slingboxServer.txt](https://github.com/GerryDazoo/Slinger/files/9864811/slingboxServer.txt)

Video of that output: https://www.youtube.com/watch?v=eZxoJjvTdCI

**Reply by GerryDazoo** [reply]
The delay is due to the buffering in the media client.

**Reply by rblancarte** [reply]
For anyone following this thread I wanted to note that this solution solved my problems with this.

**Comment by rblancarte** [comment]
One other thing, which is a big WTF for me - the 350 has built in IR blasters in the chassis?
This is what I get from not reading my manuals and just setting up like my old system! Just found this out when watching a video about setting up a slingbox.
